File: solutions/golang/atm/deposit_transaction.go
package atm

type DepositTransaction struct {
	BaseTransaction
}

func NewDepositTransaction(transactionID string, account *Account, amount float64) *DepositTransaction {
	return &DepositTransaction{
		BaseTransaction: BaseTransaction{
			TransactionID: transactionID,
			Account:       account,
			Amount:        amount,
		},
	}
}

func (t *DepositTransaction) Execute() error {
	return t.Account.Credit(t.Amount)
}

File: solutions/csharp/atm/DepositTransaction.cs
namespace ATM
{
    public class DepositTransaction : Transaction
    {
        public DepositTransaction(string transactionId, Account account, double amount)
            : base(transactionId, account, amount)
        {
        }

        public override void Execute()
        {
            account.Credit(amount);
        }
    }
}